Test mode

hazeltorek opened the pull request:
woah its test mode!!!!!!!!

when you put the robot in test mode on driver station and enable it will check which motors are prsent and put the names of all the non-present ones to a smart dashboard entry

people should update these methods when new motors are added to subsystems

hazeltorek committed:
Initial commit

* Created a new Java command-based WPILib project.

* Removed the ExampleCommand and ExampleSubsystem that are automatically
  created by the project template.

* Created a DriveSystem and DriveWithJoystick command without any
  functionality yet and declared them in RobotContainer.

hazeltorek committed:
Added Gradle task that deploys Git data

* Added two tasks to Gradle that write the current branch name and
  current short commit hash to files in src/main/deploy/.

* Made Gradle deploy task depend on the generation of those files so
  that they are created every deploy.

* Added both files to .gitignore since they are changed every deploy and
  contain data already tracked by Git.

hazeltorek committed:
Add REVLib vendordep

Added REVLib vendor dep for interfacing with CANSparkMaxes so that we can work on drivetrain

Commit:
added rev robotics vendor deps

hazeltorek committed:
Add build action

Creates a Github workflow for building code with Gradle

hazeltorek committed:
Add GitHub Actions builds

hazeltorek committed:
Added CTRE and NavX deps

wrenjaminfranklin committed:
started shooter work

hazeltorek committed:
Added CTRE and NavX deps

VZamora5 committed:
Made mecanum drivesystem and its drive command

VZamora5 committed:
Created a drive command

VZamora5 committed:
Created command to toggle field oriented

VZamora5 committed:
Documented drive command and made slow mode

VZamora5 committed:
Implemented sendable in drivesystem

Changed gitignore to include simulation files

wrenjaminfranklin committed:
fixed shootHigh to say motor 1 and 2 instead of motor 1 twice

jboutin020106 committed:
worked on the intake subsystem and associated commands

hazeltorek committed:
Started adding trajectory generation code

* This code does not work.

* Added dimensions for wheels in the DriveConstants class in Constants.

* Added placeholders for the wheelbase, track width, and kinematics
  needed for trajectory following.

* Added a method to DriveSystem for generating MecanumControllerCommands
  to follow trajectories, which takes a trajectory and a boolean for
  reversed as parameters.

* Added pose and wheel speed getters to DriveSystem.

wrenjaminfranklin committed:
wrote climb subsystem

VZamora5 committed:
Added documentation. Fixed drive command values. Renamed joystick to 'driver.'

VZamora5 committed:
Added recognition of deadzones

hazeltorek committed:
Continued trajectory following

* Added values for wheel base, track width, and a drive feedforwards
  controller to Constants.

* Declared encoders in the DriveSystem.

* Added methods for getting wheel speeds and setting the motor voltages
  in DriveSystem.

hazeltorek committed:
Merge from victoria's drivetrain branch

* Includes changes for new gyro which is needed for odometry updating.

hazeltorek committed:
Add drive system and command

jboutin020106 committed:
worked on the intake subsystem

jboutin020106 committed:
Let's see what happens (fixing git issue. . . maybe)

hazeltorek committed:
Moved PID to Spark motor controllers

* Removed the `driveVolts` method from DriveSystem and added a `mecanumDriveWheelSpeeds` method, which is needed since PID was moved to onboard motor controllers.

* Changed the MecanumControllerCommand constructor in DriveSystem to use
  the `mecanumDriveWheelSpeeds` method and not include PID for each
  motor.

* Instantiated SparkMaxPIDControllers for each motor controller with
  placeholder values.

* Moved the max speed into DriveConstants.

jboutin020106 committed:
got the RobotContainer File from the team repository

jboutin020106 committed:
got the DriveSystem file from the team repository

hazeltorek committed:
Added trajectory following as default auto

* Replaced empty autonomous with the trajectory following for an empty
  trajectory.

* Added a `getTrajectoryConfig` method to the DriveSystem for accessing
  the default configuration.

* Added a maximum acceleration constant to DriveConstants.

* Set the max speed and acceleration to 1.0 for both.

jboutin020106 committed:
implemented different variable names

hazeltorek committed:
Added motor IDs to Constants

* Added IDs for all four drive motors to Constants.

* Changed DriveSystem to have DriveConstants as a static import.

hazeltorek committed:
Added intake subsystem

Commit:
Added Limelight and PhotonVision classes. Started to write the methods and Sendable for PhotonVision -Jalen

hazeltorek committed:
better method name :)

VZamora5 committed:
Added limelight vision

VZamora5 committed:
Added photon vision

VZamora5 committed:
Added initialization of pipeline based on alliance

wrenjaminfranklin committed:
built limit switches for climb

wrenjaminfranklin committed:
added stopShooter mmethod

VZamora5 committed:
Added targeting methods to PhotonVision

VZamora5 committed:
Documented photonvision methods

jboutin020106 committed:
added a rotateToAngle method to the DriveSubsystem

VZamora5 committed:
Added outtake commands for high and low

wrenjaminfranklin committed:
added climb command

Commit:
Merge branch 'rotateToAngleBranch' of github.com:jboutin020106/frc-team342-rapid-react into driveWithTargeting

Commit:
Merge branch 'limelight' of github.com:VZamora5/frc-team342-rapid-react into driveWithTargeting

hazeltorek committed:
Trajectory following

* Added a method to the drive subsystem for creating trajectory-following commands when provided with a trajectory
* Add PID controllers and encoders to the drive system
* Added all the other methods and stuff needed for trajectory
* Changed Constants to include drive constants and motor IDs

wrenjaminfranklin committed:
added javadoc about limit switches

VZamora5 committed:
Added documentation to Outtake Subsystem

hazeltorek committed:
Outtake subsystem and commands

Subsystem for shooter and commands for shooting high and shooting low

hazeltorek committed:
Climb subsystem and commands

Climb subsystem with methods for activating and deactivating climb
Command for first level of climb

wrenjaminfranklin committed:
did dependency injection + started exitTarmac command

Commit:
Merge branch 'trajectory' of github.com:ctorek/frc-team342-rapid-react into driveWithTargeting

wrenjaminfranklin committed:
added timer for robot to run 2 or more seconds

Commit:
Adjusted my code to Connell's Pull Rq Notes

Stopped using targetingController and used rotationController (delted
hte unused targteingController). Reduced the if statement in the
isLookingLeft() method in the Limelight class.

wrenjaminfranklin committed:
changed name to "AutoDrive"

Commit:
Java docced

Java docced the methods I wrote in the last commit - Jalen

hazeltorek committed:
Merge remote-tracking branch 'jalen/driveWithTargeting' into HEAD

VZamora5 committed:
Created command to find and fulfill trajectory

hazeltorek committed:
Started test mode

* Added a test method to drive system which will be used in test mode to
  test connection to every hardware device.

jboutin020106 committed:
updated the limelight wrapper class with a method that returns the position of the robot in a 3D space

hazeltorek committed:
Reverse strafe and rotation

hazeltorek committed:
Modified drive to reflect physical robot

* Changed drive with joystick command to negate x-axis and rotation in
  order to make them the same direction as the joystick.

* Added current limiting, voltage compensation, and maximum ramp rate to
  drivesystem and constants.

VZamora5 committed:
Added deploy and retract commands

hazeltorek committed:
Started writing shooter PID

wrenjaminfranklin committed:
decleared seconds as a parameter

VZamora5 committed:
Added deploy command for intake

hazeltorek committed:
divide rotation speed by 2 when driving

hazeltorek committed:
Started adding PID loop to shooter

* Added method for checking if shooter is up to speed.

* Moved shooter speed setters to periodic and make shoot methods change
  a subsystem variable for setpoint

VZamora5 committed:
Added retract command

jboutin020106 committed:
created a method that gives the limelight access to a transform2d

hazeltorek committed:
Commands for intake deploy and retract

Intake commands and drive to cargo with photonvision

hazeltorek committed:
Shooter constants

jboutin020106 committed:
corrected the inputted values for the rotation 2d - previously was x and y, is now pitch and yaw in radians

hazeltorek committed:
3d support for Limelight

Get 3d robot position relative to vision target

jboutin020106 committed:
moved DriveToCargo to the autonomous folder

hazeltorek committed:
Added test mode results to RobotContainer

* Finished drive system test method

* Added a RobotContainer method to send the test results to the
  dashboard

hazeltorek committed:
Fix test mode in drive

* Changed drive test method to use .equals() for strings

* Added a call to the test results method in Robot in testInit

hazeltorek committed:
AutoDrive command 

command for driving forward at a set speed for a specified amount of time, used for exiting tarmac in auto

jboutin020106 committed:
creates the command group that allows the robot to drive to the hub

VZamora5 committed:
Programmed autonomous command; Drivefunctions

jboutin020106 committed:
Merge branch 'frc-team-342:dev' into driveToHub

jboutin020106 committed:
imported the sequential command group

jboutin020106 committed:
Merge branch 'driveToHub' of github.com:jboutin020106/frc-team342-rapid-react into driveToHub

hazeltorek committed:
Made motor names in drive more specific

jboutin020106 committed:
fixed limelight to use NumberArray

hazeltorek committed:
Merge remote-tracking branch 'team/dev' into test-mode

hazeltorek committed:
Added test for shooter motors

hazeltorek committed:
Command for driving to LL target

Used to drive to the limelight target on the hub using a trajectory

hazeltorek committed:
Added test for climb and intake

hazeltorek committed:
Created Constants for climb and intake

* Created climb constants and changed climb subsystem to reference them.

* Created intake constants and changed intake subsystem to reference
  them.

* Did not make outtake constants because they are on the shooter-pid
  branch.

JalenEvans reviewed (approved):
I think it looks grand! Dictionaries are better tho.

jboutin020106 committed:
Added the Shoot3Start Auto, added speed parameter to AutoDrive,added a RotateToAngle command

VZamora5 committed:
Renamed "Intake" to "intake"

jboutin020106 committed:
fixed extraneous imports

hazeltorek committed:
Autonomous that shoots 3 cargo (#17)

Added an AutoDrive constructor that takes a speed and an autonomous command group that shoots 3 cargo

hazeltorek committed:
javadoc test methods

hazeltorek committed:
javadoc main test method

hazeltorek committed:
Added tests for limelight and photonvision

hazeltorek committed:
only feed when shooter up to speed

hazeltorek committed:
instantiated outtake command and subsystem

* Fixed OuttakeHigh to dependency inject correctly

* Added the outtake subsystem and outtake high command to robotcontainer

jboutin020106 committed:
switched the talonSRX motors to talonFX motors

hazeltorek committed:
Make outtake subsystem into sendable

hazeltorek committed:
Fix photon and limelight tests

* Changed photon vision test to check if latency is not equal to zero
  milliseconds

* Changed limelight and photon tests to fail on value being incorrect

hazeltorek committed:
Changes for physical robot

hazeltorek committed:
oops import cringe

hazeltorek committed:
fixed comment about rpm conversion

hazeltorek committed:
commented about statically imported constants

jboutin020106 committed:
Wrote the appropriate methods to control the second stage climber

VZamora5 committed:
Implemented getDistance method in limelight

VZamora5 committed:
Implemented getDistance to limelight

VZamora5 committed:
Implemented a getDistance method for photon vision

jboutin020106 committed:
deleted unecessary imports and adjusted the associated methods to have concise if statements

jboutin020106 committed:
Merge branch 'frc-team-342:dev' into srxToFx

jboutin020106 committed:
removed an unused scanner import

hazeltorek committed:
Second stage of climb (#22)

:)

hazeltorek committed:
Changed encoder tick to rpm conversion

VZamora5 committed:
Made fixes

hazeltorek committed:
removed navx dependency

hazeltorek committed:
fix comment grammar

hazeltorek committed:
Merge branch 'dev' into shooter-pid

hazeltorek committed:
PID loop for shooter (#18)

VZamora5 committed:
Fixed target measurements

VZamora5 committed:
Fixed package name from Intake to intake

hazeltorek committed:
remove empty deploy command

MatthewEdgmon committed:
Actually fix package name

hazeltorek committed:
Fix frc.robot.commands.intake package name (#23)

hazeltorek committed:
Merge branch 'dev' into driveFunctions

VZamora5 committed:
Fix merge conflict

VZamora5 committed:
Fix merge conflict 2

VZamora5 committed:
Fix build

hazeltorek committed:
Drive functions (#20)

JalenEvans reviewed (approved):
Black Magic is used but the rest is good!

hazeltorek committed:
Merge branch 'dev' into test-mode
